<?xml version="1.0"?>
<launch>
    <arg name="gripper_port_name" default="/dev/q_gripper_r"/>
    <arg name="launch_fjtas" default="true"/>
    <arg name="gripper_name" default="q_gripper"/>
    <arg name="joint_offset" default="1900"/>
        
    <!-- the driver node, talking to the gripper using the specifed port and (virtual) baudrate -->
    <node pkg="q_gripper" type="q_gripper_driver.py" name="q_gripper" output="screen" >
        <param name="rate_hz" value="1000" />
        <param name="baud_rate" value="115200" />
        <param name="port_name" value="$(arg gripper_port_name)" />
        <param name="verbose" value="1" />
        <param name="hand_name" value="$(arg gripper_name)" />
        <param name="joint_offset" value="$(arg joint_offset)" />
        <remap from="q_gripper/joint_states" to="/joint_states" />
    </node>

    <!-- the gripper FollowJointTrajectoryAction server (e.g. needed for Moveit) -->
    <node pkg="q_gripper" type="q_gripper_fjtas.py" name="q_gripper_fjtas" output="screen" if="$(arg launch_fjtas)">
        <remap from="q_gripper_fjtas/joint_states" to="/joint_states" />
        <remap from="q_gripper_fjtas/joint_goals" to="/q_gripper/joint_goals" />
        <param name="hand_name" value="$(arg gripper_name)" />
    </node>

</launch>
